app/src/main/kotlin/com/simplemobiletools/calculator/helpers/MoneyCalculatorImpl.kt

```diff
@@ -0,0 +1,87 @@
+package com.simplemobiletools.calculator.helpers
+
+import android.content.Context
+import com.simplemobiletools.calculator.R
+
+class MoneyCalculatorImpl(calculator: Calculator, val context: Context) {
+    private var mCallback: Calculator? = calculator
+
+    private var number: String = ""
+    private var decimalClicked: Boolean = false
+    private var decimalCounter: Int = 0
+
+    init {
+        setValue("0.00")
+        handleClear()
+    }
+
+    fun numpadClicked(id: Int) {
```

Comment:
Maybe we can refactor this function up into Calculator and make Calculator an abstract class to reduce duplication.

Reply:
Calculator is used by the activities not the Impl classes currently. Kinda weird how this was built. I just built like the other activity.

Reply:
I'm not sure that would work since the number being displayed here is a String whilst it's a float in calculatorImpl.

Reply:
I think we should stick with the interface. we could refactor the switch to a helper class with a static method in the future. I don't know how this would lower code complexity. Also, we are not I do not think we are going to add new calculator activities using code like this. We have this code repeated twice, but I don't think it necessarily needs to be refactored at this point.

Reply:
Alright fair enough 🛫